### PR TITLE
feat: add user repository

### PR DESCRIPTION
**Pull request by mabdh**
rebased on https://github.com/odpf/columbus/pull/65

will assign to be reviewed once https://github.com/odpf/columbus/pull/65 has got merged.

### CODE REVIEW

File: user/service.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package user
+
+import "context"
+
+// Service is a type of service that manages business process
+type Service struct {
```
Comment (StewartJingga) [inline user/service.go:6]: If there is no business logic, should we skip Service struct and go straight to UserRepo? wdyt?
We can always introduce service layer when business logic arises.

Im still thinking if it is a good idea to create service just for the sake of having a service. 🤔

File: user/service.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	repository Repository
+}
+
+// Create handles create business operation for user
```
Comment (StewartJingga) [inline user/service.go:10]: wrong comment, might've missed it.

**Review by StewartJingga — CHANGES_REQUESTED** [review]

File: user/service.go (hunk 1 above)
Comment (mabdh) [reply user/service.go:6]: agree with this

File: user/service.go (hunk 2 above)
Comment (mabdh) [reply user/service.go:10]: this will be removed in this PR

**Review by StewartJingga — APPROVED** [review]
LGTM

**Pull request merged**